src/interop/js-interop/js-types.md

```diff
@@ -0,0 +1,242 @@
+---
+title: JS types
+description: Usage information about the core types in JS interop.
+---
+
+Dart values and JS values belong to separate language domains. When compiling to
+Wasm, they execute in separate *runtimes* as well. As such, you should treat JS
+values as foreign types. To provide Dart types for JS values,
+[`dart:js_interop`] exposes a set of [extension types] prefixed with `JS` called
+"JS types".
+
+Importantly, these types have compiler-specific [representation types]. This
```

Comment:
Consider shuffling how we present the info in this paragraph, to be more upfront about what this means for the reader/developer, and if we find that it's necessary we can explain the rationale.

I think that the "representation types" is the reason behind things, but what matters to developers is that these JS types are not really reified at runtime and not even consistent across implementations.

A possible suggestion:
> JS types are used to help developers and tools track, at compile-time, when the program is operating over Dart or JS values. However, at runtime, JS types are not fully reified and their implementation is different when you compile to Wasm or JS. This imposes important [restrictions](#restrictions) of how to use these values: they cannot be used in is-tests and casts like regular Dart types. You can, however, work with them through conversions or through interop calls.

Reply:
Reworded to take into account your feedback, highlighting the motivation, the difference in runtime semantics, and the possibility of using conversions.